### DISCUSSION TITLE
On-connect event settings (to invert stepper)

### DISCUSSION THREAD

**Discussion by cdibbs** [arkypita/LaserGRBL#1627] [Ideas]
Sorry in advance if I missed another discussion which already covered this.

Each time I connect to my laser cutter, I have to remember to invert the step direction. This is just a preference on my part due to how I've oriented my machine in my room. But my laser cutter forgets this setting each time the machine is powered off. Does it seem like a good idea to add a GCode setting for the "on connected" event which automatically sends initialization gcode, in my case `$3=3`? Is there another way to accomplish this? If not, would you like me to do a PR for it? Thanks!

**Comment by arkypita** [comment]
It's a very strange behavior, grbl should keep all configured parameters in controller flash memory. If it doesn't, either you have an unofficial version of grbl (perhaps customized by your engraver manufacturer) or you have some initialization code stored in the controller.

> Does it seem like a good idea to add a GCode setting for the "on connected" event which automatically sends initialization gcode, in my case $3=3? Is there another way to accomplish this?

You can get what you ask for without any LaserGRBL intervention, using grbl startup blocks.

https://github.com/gnea/grbl/wiki/Grbl-v1.1-Commands#n---view-startup-blocks

**Reply by cdibbs** [reply]
@arkypita This looked promising but when I try it only responds with "Expected command letter." I also have not been able to find any references on how to do it. How do you escape `$3=3` so that $N0 will accept it in the assignment: `$N0=$3=3`? I've tried a few common escapes and using parentheses and brackets to no avail. Although, `$N0=($3=3)` at least did not error. However, it also didn't echo back the value when I typed `$N`.

**Comment by cdibbs** [comment]
I never could get this working via startup blocks. Instead, I serendipitously resolved it by updating the firmware to the generic v1.1h provided by LaserGRBL _and restoring settings from before the update._
